File: 2021/6/solutions/perl/slate/day6.pl
my @input = ();

while(<>) {
	push @input, $_;
}

my @in  = split(',' , @input[0]);

my @fish = (); $#fish = 9;

foreach (@in){
	@fish[$_]++;
}

my $counter = 0;
my $counter2 = 0;

for ($j = 0; $j < 80; $j++){
	$tmp = @fish[0];
	$length = (scalar @fish) - 1;
	for ($i = 0; $i <= $length; $i++){
		@fish[$i] = @fish[$i+1];
	}
	@fish[6] += $tmp;
	@fish[8] = $tmp;
}

foreach (@fish){
	$counter += $_;
}

print $counter, "\n";
$j = 0;

my @fish = (); $#fish = 9;
foreach (@in){
	@fish[$_]++;
}

for ($j = 0; $j < 256; $j++){
	$tmp = @fish[0];
	$length = (9) - 1;
	for ($i = 0; $i <= 9; $i++){
		@fish[$i] = @fish[$i+1];
	}
	@fish[6] += $tmp;
	@fish[8] = $tmp;
}

foreach (@fish){
	$counter2 += $_;
}

print $counter2, "\n";

File: 2021/6/solutions/python/slate/day6.py
import sys

input = sys.stdin.readlines()

inp = input[0].split(',')

fish = [0] * 9

for i, val in enumerate(inp):
	intval = int(val)
	fish[intval]+=1

counter = 0
counter2 = 0

for j in range(80):
	tmp = fish[0]
	for ii in range(len(fish) - 1):
		fish[ii] = fish[ii+1]
	fish[6] += tmp
	fish[8] = tmp

for fi in fish:
	counter += fi

print(counter)

fish = [0] * 9

for i, val in enumerate(inp):
	intval = int(val)
	fish[intval] += 1

for j in range(256):
	tmp = fish[0]
	for ii in range(len(fish) - 1):
		fish[ii] = fish[ii+1]
	fish[6] += tmp
	fish[8] = tmp

for fi in fish:
	counter2 += fi

print(counter2)
